### PR TITLE
feat: support graphQL query for templates with filters

### PR DESCRIPTION
**Pull request by HegelPro**
affects: @amplicode/codegen

### CODE REVIEW

File: example-app/src/app/screens/owner/OwnerTableWithFilter.tsx [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+        </Col>
+      </Row>
+
+      <Space>
+        <Button type="primary" htmlType="submit">
+          <FormattedMessage id="filters.apply" />
+        </Button>
+        <Button onClick={() => form.resetFields()}>
+          <FormattedMessage id="filters.reset" />
+        </Button>
```
Comment (vyacheslav-pushkin) [inline example-app/src/app/screens/owner/OwnerTableWithFilter.tsx:243]: I think it better to rename "Clear" -> "Reset".
The button should not only clear the fields, but also reset the items selection (i.e apply).

File: example-app/src/app/screens/owner/OwnerTableWithFilter.tsx [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+      <Row gutter={16}>
+        <Col span={8}>
+          <Form.Item label="First Name" name={"firstName"}>
+            <Input />
+          </Form.Item>
+        </Col>
+
+        <Col span={8}>
+          <Form.Item label="Last Name" name={"lastName"}>
+            <Input />
```
Comment (vyacheslav-pushkin) [inline example-app/src/app/screens/owner/OwnerTableWithFilter.tsx:232]: Let's add "clear" icon that will clear the given input (but will not apply the filters). The icon should be the same to one used in EntityLookupField.

**Review by vyacheslav-pushkin — CHANGES_REQUESTED** [review]

File: example-app/src/app/screens/owner/OwnerTableWithFilter.tsx [hunk 3]
```diff
@@ -0,0 +1,308 @@
+import { useEffect, useState } from "react";
```
[the remark this answers is not in the document]
Comment (HegelPro) [reply example-app/src/app/screens/owner/OwnerTableWithFilter.tsx:1]: We need to support petList query with filter in [amplicode-mvp-petclinic](https://github.com/Amplicode/amplicode-mvp-petclinic). Now is supported query with filter just for Owner entity.
I can add filter logic to petList but filters will not applying until backend supporting in [amplicode-mvp-petclinic](https://github.com/Amplicode/amplicode-mvp-petclinic) project.

File: example-app/src/app/screens/owner/OwnerTableWithFilter.tsx (hunk 1 above)
Comment (vyacheslav-pushkin) [reply example-app/src/app/screens/owner/OwnerTableWithFilter.tsx]: > The button should not only clear the fields, but also reset the items selection (i.e apply).

This is not done.

File: example-app/src/app/screens/owner/OwnerTableWithFilter.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (vyacheslav-pushkin) [reply example-app/src/app/screens/owner/OwnerTableWithFilter.tsx]: We need entity-management examples both with and without filters. Therefore please create the following screens:

- Owner Table (no filters)
- Owner Cards (no filters)
- Owner List (no filters)
- new Owner Cards (with filters)
- Pet Table (with filters)
- Pet Cards (with filters)
- Pet List (with filters)

File: packages/codegen/src/common/studio/studio-model.ts [hunk 4]
```diff
@@ -19,6 +19,8 @@ export const enum StudioTemplatePropertyType {
   ROUTE = 'ROUTE',
   ATTRIBUTE = 'ATTRIBUTE',
   ATTRIBUTES_ARRAY = 'ATTRIBUTES_ARRAY',
+  ARGUMENT = 'ARGUMENT',
```
Comment (vyacheslav-pushkin) [inline packages/codegen/src/common/studio/studio-model.ts:22]: Please remove `ARGUMENT`. Every argument we add must be supported by Studio, but this one is not used anywhere yet.

File: packages/codegen/src/common/studio/studio-model.ts [hunk 5]
```diff
@@ -19,6 +19,8 @@ export const enum StudioTemplatePropertyType {
...
+  ARGUMENT_ARRAY = 'ARGUMENT_ARRAY',
```
Comment (vyacheslav-pushkin) [inline packages/codegen/src/common/studio/studio-model.ts:23]: Let's rename it to `FILTER_QUERY_ARGUMENT_ARRAY`. We will likely want Studio to exclude arguments related to pagination and sorting, so I think we need a less generic / more semantic name.

File: packages/codegen/src/generators/react-typescript/entity-list/answers.ts [hunk 6]
```diff
@@ -70,9 +70,9 @@ export const entityListQuestions: StudioTemplateProperty[] = [
...
   {
-    caption: "Select attributes to filter by",
-    code: "filterByAttributes",
-    propertyType: StudioTemplatePropertyType.ATTRIBUTES_ARRAY,
+    caption: "Select arguments to filter by",
+    code: "filterByArguments",
+    propertyType: StudioTemplatePropertyType.ARGUMENT_ARRAY,
     relatedProperty: 'query',
     required: false,
     defaultValue: [],
```
Comment (vyacheslav-pushkin) [inline packages/codegen/src/generators/react-typescript/entity-list/answers.ts:78]: Please remove `defaultValue`.

File: scripts/generate-management/owner-management-table.js [hunk 7]
```diff
@@ -1,16 +1,17 @@
...
     itemComponentName: 'OwnerTableEditor',
     route: 'owner-table',
     type: 'table',
     shouldAddToMenu: true,
-    listQuery: esc(ownerListQuery),
+    listQuery: esc(ownerListQueryWithFilter),
     detailsQuery: esc(ownerDetailsQuery),
     deleteMutation: esc(ownerDeleteMutation),
     upsertMutation: esc(ownerUpsertMutation),
+    filterByArguments: [['filter', 'firstName'], ['filter', 'lastName']]
```
Comment (vyacheslav-pushkin) [inline scripts/generate-management/owner-management-table.js:14]: Let's add the tests for the following casses:

- `filterByArguments` is `null`
- `filterByArguments` is `undefined`
- `filterByArguments` is empty array

For all these cases we need to generate a screen without filters.

**Review by vyacheslav-pushkin — APPROVED** [review]
Please fix comments and merge

**Pull request merged**